src/backend/metal/src/command.rs

```diff
@@ -1495,29 +1555,24 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
             depth: 1,
         };
 
-        let commands = vec![
+        let commands = [
             soft::ComputeCommand::BindPipeline(pso),
             soft::ComputeCommand::BindBuffer {
                 index: 0,
-                buffer: Some(buffer.raw.clone()),
+                buffer: Some(&*buffer.raw),
                 offset: start,
             },
             soft::ComputeCommand::BindBufferData {
                 index: 1,
-                bytes: unsafe {
-                    slice::from_raw_parts(
-                        value_and_length.as_ptr() as _,
-                        mem::size_of::<u32>() * value_and_length.len()
-                    ).to_owned()
-                },
+                words: &value_and_length[..],
             },
             soft::ComputeCommand::Dispatch {
                 wg_size,
                 wg_count,
             },
         ];
 
-        inner.sink().begin_compute_pass(commands);
+        inner.sink().begin_compute_pass(commands.iter().cloned());
```

Comment:
Seems like we should be able to avoid `cloned` here?

Reply:
Actually, no. The iterator of a slice is over `&Command` while we expect commands by move there. Cloning is lightweight, since this are commands over references.